File: basic.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    var buffer: [100]u8 = undefined;
    
    // Print welcome message (like BASIC's PRINT)
    try stdout.print("BASIC-LIKE PROGRAM IN ZIG\n", .{});
    
    // Input name (like BASIC's INPUT)
    try stdout.print("WHAT IS YOUR NAME? ", .{});
    const stdin = std.io.getStdIn().reader();
    if (try stdin.readUntilDelimiter(&buffer, '\n')) |name| {
        // FOR loop (like BASIC's FOR/NEXT)
        try stdout.print("\nCOUNTING FOR {s}:\n", .{name});
        var i: i32 = 1;
        while (i <= 5) : (i += 1) {
            try stdout.print("{d}\n", .{i});
            // Simulate BASIC's delay
            std.time.sleep(500 * 1000 * 1000);
        }
        
        // IF/THEN style condition
        if (name.len > 5) {
            try stdout.print("\n{s} IS A LONG NAME!\n", .{name});
        } else {
            try stdout.print("\n{s} IS A SHORT NAME!\n", .{name});
        }
    }
} 

File: basic.cpp
#include <iostream>
#include <string>
#include <thread>
#include <chrono>

int main() {
    std::string name;
    
    // Print welcome message (like BASIC's PRINT)
    std::cout << "BASIC-LIKE PROGRAM IN C++\n";
    
    // Input name (like BASIC's INPUT)
    std::cout << "WHAT IS YOUR NAME? ";
    std::getline(std::cin, name);
    
    // FOR loop (like BASIC's FOR/NEXT)
    std::cout << "\nCOUNTING FOR " << name << ":\n";
    for (int i = 1; i <= 5; i++) {
        std::cout << i << "\n";
        // Simulate BASIC's delay
        std::this_thread::sleep_for(std::chrono::milliseconds(500));
    }
    
    // IF/THEN style condition
    if (name.length() > 5) {
        std::cout << "\n" << name << " IS A LONG NAME!\n";
    } else {
        std::cout << "\n" << name << " IS A SHORT NAME!\n";
    }
    
    return 0;
} 